onmt/tests/test_beam_search.py

```diff
@@ -290,13 +290,8 @@ def test_beam_is_done_when_n_best_beams_eos_using_min_length(self):
                     beam_idx = min(beam_sz - 1, k)
                     word_probs[beam_idx::beam_sz, j] = score
             else:
-                word_probs[0::beam_sz, eos_idx] = valid_score_dist[0]
-                word_probs[1::beam_sz, eos_idx] = valid_score_dist[0]
-                # provide beam_sz other good predictions in other beams
-                for k, (j, score) in enumerate(
-                        zip(_non_eos_idxs, valid_score_dist[1:])):
-                    beam_idx = min(beam_sz - 1, k)
-                    word_probs[beam_idx::beam_sz, j] = score
+                for j in range(beam_sz):
+                    word_probs[j::beam_sz, eos_idx] = valid_score_dist[0]
```

Comment:
I think the purpose of the test is not the same anymore

Reply:
well yes and no, but just the title of the test is no longer relevant "done when n_best beams" which is no longer the reality since we need to "beam.done" when we have beam size hyps. This is the reason why I force all beams to finish in tihs place.

Reply:
Shall we change the name of the test then? :)